go/ai/option.go

```diff
@@ -430,6 +446,20 @@ func WithOutputFormat(format OutputFormat) OutputOption {
 	return &outputOptions{OutputFormat: format}
 }
 
+// WithOutputInstructions sets whether output instructions should be added.
+func WithOutputInstructions(addInstructions bool) OutputOption {
```

Comment:
This is supposed to take in a string, which is the user's desired instructions to use in lieu of the default instructions. We store it as a pointer to a string. If it's nil, that means the user wants default instructions. If it's an empty string (not nil), the user does not want instructions to be added. If it's not an empty string, we use that as the instructions.